### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by fritz-astronomer**
- Binary Packaging (extra deployment ability)
- Helm Values via Kube Secrets (remove helm requirement) 
- add `kubectl` mode for `--kubernetes`
- Add Retrying to Kubernetes methods
- Change Progress Bar to `halo` package
- add telescope version, report date, organization name fields
- remove excess to support thin client (migrated to Airflow DAG)

### CODE REVIEW

File: telescope/functions/astronomer_enterprise.py [hunk 1]
```diff
@@ -56,6 +72,46 @@ def verify():
...
+                except Exception as e:
+                    log.exception(e)
+                output[namespace] = {
+                    "first_deployed": helm_contents["info"]["first_deployed"],
+                    "last_deployed": helm_contents["info"]["last_deployed"],
+                    "chart_metadata": helm_contents["chart"]["metadata"],
+                    "chart_values": helm_contents["chart"]["values"],
+                }
+        except Exception as e:
+            log.debug(e)
```
Comment (frankcash) [inline telescope/functions/astronomer_enterprise.py:111]: why is this exception only using debug log level?

File: telescope/functions/autodiscover.py [hunk 2]
```diff
@@ -51,25 +53,44 @@ def get_kubernetes_uniqueness(r: "kubernetes.clients.models.V1Pod"):
...
-    if kube_client is not None:
-        seen_uniqueness = set()
-        results = []
+    seen_uniqueness = set()
+    results = []
+    if os.getenv("TELESCOPE_KUBERNETES_METHOD", "") == "kubectl":
+        jp = (
+            """'{range .items[*]}{.metadata.name}{","}{.metadata.namespace}{","}{.metadata.generate_name}{"||"}{end}'"""
+        )
+        cmd = "kubectl get pod -l " + label_selector + " -A -o=jsonpath=" + jp
```
Comment (frankcash) [inline telescope/functions/autodiscover.py:62]: would be cleaner to make this an f string

File: telescope/getters/kubernetes.py [hunk 3]
```diff
@@ -30,32 +32,47 @@ def get(self, cmd: Union[List[str], str]) -> Union[dict, str]:
...
-            command=cmd,
-            container=self.container,
-            stderr=True,
-            stdin=False,
-            stdout=True,
-            tty=False,
-        )
-        log.debug(f"Got output: {exec_res}")
+            cmd = f"kubectl exec -it -n {self.namespace} {self.name} -c {self.container} -- {cmd}"
+            print(cmd)
```
Comment (frankcash) [inline telescope/getters/kubernetes.py:40]: should this be a log

File: telescope/getters/kubernetes.py [hunk 4]
```diff
@@ -30,32 +32,47 @@ def get(self, cmd: Union[List[str], str]) -> Union[dict, str]:
...
+            out = run(cmd, hide=True, warn=True).stdout  # other options: timeout, warn
+            out = clean_airflow_report_output(out)
```
Comment (frankcash) [inline telescope/getters/kubernetes.py:42]: why do you override out like this, might be more readable to use different variable names here

File: telescope/getters/kubernetes.py [hunk 5]
```diff
@@ -30,32 +32,47 @@ def get(self, cmd: Union[List[str], str]) -> Union[dict, str]:
...
+            return out
```
Comment (frankcash) [inline telescope/getters/kubernetes.py:43]: Functions generally are easier to understand with one exit point (`return`), is there a reason you chose to structure this function the way you did?

File: telescope/functions/autodiscover.py (hunk 2 above)
Comment (fritz-astronomer) [reply telescope/functions/autodiscover.py:64]: agreed - I had some `{`'s in the `jp` above, which is why I originally didn't

File: telescope/getters/kubernetes.py (hunk 5 above)
Comment (fritz-astronomer) [reply telescope/getters/kubernetes.py]: Good point - moved a few functions to only return once.

**Pull request merged**